names/names.py

```diff
@@ -13,16 +14,26 @@
 duplicates = []  # Return the list of duplicates in this data structure
 
 # Replace the nested for loops below with your improvements
-for name_1 in names_1:
-    for name_2 in names_2:
-        if name_1 == name_2:
-            duplicates.append(name_1)
+# first_name = names_1[0]
 
-end_time = time.time()
-print (f"{len(duplicates)} duplicates:\n\n{', '.join(duplicates)}\n\n")
-print (f"runtime: {end_time - start_time} seconds")
+# bst = BinarySearchTree(first_name)
+
+# after_first_in_names1 = names_1[1:]
+
+# [bst.insert(name) for name in after_first_in_names1]
+# [duplicates.append(duplicated_name) for duplicated_name in names_2 if bst.contains(duplicated_name)]
+
```

Comment:
Something to remember here is that any time you use a list comprehension, you are creating a new list. So in this case you're creating two new O(n) space lists that look like `[None, None, None, None, None, None, ...etc]` which could sit in memory until the program terminates. Garbage collection _might_ pick up the unused arrays but I wouldn't rely on it. In general, it's best to reserve list comprehensions for creating new lists and stick to for loops for your day-to-day iteration needs.